lesson16/src/js/main.js

```diff
@@ -0,0 +1,205 @@
+		const articleData = category.article;
+		tabContentsArea.appendChild(newsContentArea).appendChild(renderNewsArticle(articleData));
+	})
+}
+
+function renderCategoryTab(data){
+	const fragment = document.createDocumentFragment();
+
+	data.forEach((target,i) => {
+		const categoryTab = document.createElement('li');
+		categoryTab.classList.add('tab_list_item');
+		categoryTab.setAttribute('role','presentation');
+
+		const categoryButton = document.createElement('button');
+		categoryButton.textContent = target.category;
+		categoryButton.setAttribute('id','tab-' + (i+1));
+		categoryButton.setAttribute('class', 'tab-button js-tabButton');
+		categoryButton.setAttribute('aria-controls','panel-' + (i+1));
+		categoryButton.setAttribute('role', 'tab');
+		categoryButton.setAttribute('aria-selected', 'false');
+		categoryButton.setAttribute('tabindex', '-1');
+		fragment.appendChild(categoryTab).appendChild(categoryButton);
+	});
+	const categoryTabLists = document.querySelector('.tab-list');
+	categoryTabLists.appendChild(fragment);
+}
+
+const renderNewsArticle = (data) => {
+	const newsLists = document.createElement('ul');
+	newsLists.classList.add('news-list');
+	const fragment = document.createDocumentFragment();
+	const displayedResults = 5;
+	for (let i = 0; i < displayedResults; ++i){
+		const li = document.createElement('li');
+		li.classList.add('news-item');
+
+		const anchor = document.createElement('a');
+		anchor.classList.add('news-link');
+		anchor.href = data[i].link;
+		anchor.textContent = data[i].title;
+		fragment.appendChild(li).appendChild(anchor);
+		addComment(data[i], li);
+		addNewIcon(data[i],li);
+	}
+	newsLists.appendChild(fragment);
+	return newsLists;
+};
+
+function tabContentsInitialDisplay(data){
+	data.forEach((target) =>{
+		const activeCategory = target.isActive;
+
+		if(activeCategory){
+			const categoryButton = categoryTab.querySelector('.js-tabButton');
+			categoryButton.setAttribute('tabindex', '0');
+			categoryButton.setAttribute('aria-selected','true');
+
+			const getSelectedTadID = categoryButton.getAttribute('aria-controls');
+			const selectedPanel = document.getElementById(getSelectedTadID);
+			selectedPanel.setAttribute('aria-hidden', false);
+		}
+	})
+}
+
+function addNewsImage(data){
+	const getNewsArea = document.querySelectorAll('.news-area');
+
+	data.forEach((category,i) => {
+		const imageArea = document.createElement('div');
+		const img = document.createElement('img');
+		img.src = category.image;
+		img.width = 200;
+		img.height = 200;
+		getNewsArea[i].appendChild(imageArea).appendChild(img);
+	})
+
+}
+
+function addComment(data, item) {
+	const commentResult = data.comment;
+	if (commentResult.length) {
+		const commentBox = document.createElement('span');
+		const anchor = document.createElement('a');
+		anchor.classList.add('comment-count');
+		anchor.textContent = commentResult.length;
+		item.appendChild(commentBox).appendChild(anchor);
+	}
+}
```

Comment:
IMO
It is recommended that split assignments be used. Also, it is better to assign to variables what you really use!
```suggestion
function addComment({ comment }, item) {
  const commentResultLength = comment.length;
  if (commentResultLength) {
    const commentBox = document.createElement("span");
    const anchor = document.createElement("a");
    anchor.classList.add("comment-count");
    anchor.textContent = commentResultLength;
    item.appendChild(commentBox).appendChild(anchor);
  }
}
```

Reply:
I have a pritter, but it seems to be strange on GitHub, so when I checked it, there was a problem with the VScode setting.
I changed the setting referring to the following and uploaded it again.
https://qiita.com/pyon_kiti_jp/items/031de3fdd3c60ddc21bc

[2e40b0e](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/2e40b0eb0c5907b388461a6fbf102791f55892bc)

Reply:
すみません、私のお伝えしたことが英語だとお伝えしきれていなかったようです。
以下は
> It is recommended that split assignments be used. Also, it is better to assign to variables what you really use!
分割代入をつかって、本当に使うものを変数に代入したほうがいいと思います！

とコメントしたつもりでした。（スペースのことをお話したわけではありませんでした。設定を変更させてしまいすみません…！）
なので上記の通り
```js
//ここでdataを分割代入しています。
function addComment({ comment }, item) {
//data.comment ではなくdata.comment.lengthを代入しています。
  const commentResultLength = comment.length;
  if (commentResultLength) {
    const commentBox = document.createElement("span");
    const anchor = document.createElement("a");
    anchor.classList.add("comment-count");
    anchor.textContent = commentResultLength;
    item.appendChild(commentBox).appendChild(anchor);
  }
}
```
としたほうが良いというIMOなのですが、こちらについてはどう思いますか？

Reply:
I'm terribly sorry!
I didn't understand the split assignment.
I didn't really understand split assignment, and your review helped me to understand it!

I have fixed it.

[169a767](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/169a767f5360cc71b9a1e3ace5e2dd36d1f52893)